server/storage/schema/auth_users.go

```diff
@@ -31,10 +32,7 @@ func (atx *authBatchTx) UnsafeGetUser(username string) *authpb.User {
 	return arx.UnsafeGetUser(username)
 }
 
-func (abe *authBackend) GetAllUsers() []*authpb.User {
-	tx := abe.BatchTx()
-	tx.Lock()
-	defer tx.Unlock()
+func (abe *authBackend) GetAllUsers(tx auth.AuthReadTx) []*authpb.User {
```

Comment:
I would suggest to remove `GetAllUsers` and users call `UnsafeGetAllUsers` directly. All methods prefixed with `Unsafe` are unsafe operations, otherwise, they are safe operations.